### ISSUE TITLE
[Travis] hashie gem is causing an error

### ISSUE THREAD

**Issue by tmiotto** [HewlettPackard/oneview-chef#163]
### Scenario/Intent
Travis is not executing unit tests due to some kind of breakage involving `chefspec` gem.

You can see the master last commit (that passed in a previous Travis check) [failing right here in this new branch](https://travis-ci.org/HewlettPackard/oneview-chef/builds/197030113).

### Environment Details
- **OneView Cookbook version:** 1.10
 - **platform:** https://travis-ci.org/

### Steps to Reproduce
Push any commit to this remote.

### Expected Result
Travis should perform the unit tests and output if it was approved or not.

### Actual Result
Travis is actually breaking and not approving any commit.

**Comment by jsmartt** [comment]
The only difference between gem versions in a recent build that passed and one of the new failing builds is the hashie gem (3.4.6 -> 3.5.0). I think there's a bug in v3.5.0 where a file isn't required before it's used. That or whatever is using hashie didn't require it correctly.

Looks like there's an issue posted [here](https://github.com/intridea/hashie/issues/391)

**Comment by jsmartt** [comment]
Fixed with hashie release 3.5.1

**Comment by jsmartt** [comment]
So we're no longer getting an error, but we are still getting a huge list of warnings with that new hashie gem. We'll need to keep an eye out; hopefully something will change with either hashie or chefspec to get rid of these errors:
```
W, [2017-01-31T21:08:22.921088 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#frozen? defined in Kernel. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:22.921274 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#frozen? defined in Kernel. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:22.921424 #16306]  WARN -- : You are setting a key that conflicts with a built-in method VariaModel::Attributes#frozen? defined in Kernel. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:22.921551 #16306]  WARN -- : You are setting a key that conflicts with a built-in method VariaModel::Attributes#frozen? defined in Kernel. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:22.927597 #16306]  WARN -- : You are setting a key that conflicts with a built-in method VariaModel::Attributes#default defined in Hash. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:22.928040 #16306]  WARN -- : You are setting a key that conflicts with a built-in method VariaModel::Attributes#default defined in Hash. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:29.946394 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:29.947364 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:29.948147 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:30.347119 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:33.716072 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:33.716984 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:33.717787 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:33.718591 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:33.719396 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:33.720232 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.027364 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.028408 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.029199 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.030072 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.030923 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.031785 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.032638 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:35.033474 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:36.466588 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#fetch defined at /home/travis/.rvm/gems/ruby-2.3.1/gems/hashie-3.5.1/lib/hashie/mash.rb:141. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
W, [2017-01-31T21:08:36.922756 #16306]  WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#zip defined in Enumerable. This can cause unexpected behavior when accessing the key via as a property. You can still access the key via the #[] method.
```

**Comment by michaelherold** [comment]
I just released Hashie v3.5.2 which has the `disable_warnings` feature for the Mash logger and sets the Hashie logger to the Rails logger when we detect that we're running in Rails. Hopefully, this helps!
